src/control/src/pure_pursuit/pp.cpp

```diff
@@ -0,0 +1,144 @@
+#include "pure_pursuit/pp.hpp"
+
+/**
+ * @brief Pure Pursuit class Constructor
+ *
+ * @param k Lookahead gain
+ */
+
+PurePursuit::PurePursuit(double k, double ld_margin) : k_(k), ld_margin_(ld_margin) {}
+
+double PurePursuit::update_steering_angle(
+    const custom_interfaces::msg::PathPointArray::ConstSharedPtr &path_msg,
+    const custom_interfaces::msg::Pose::ConstSharedPtr &pose_msg) {
+  update_vehicle_pose(pose_msg);
+
+  this->ld_ = this->update_lookahead_distance();
+
+  // update closest point
+  auto closest_point_info = this->update_closest_point(path_msg);
+  this->closest_point_ = closest_point_info.first;
+  this->closest_point_id_ = closest_point_info.second;
+
+  // update lookahead point
+  auto lookahead_point_info = this->update_lookahead_point(path_msg);
```

Comment:
May be easier for testing if this receives the closest point as argument, but just a suggestion

Reply:
on it!